### PR TITLE
1984: Suppressed Bans

### PR DESCRIPTION
**Pull request by francinum**
closes #5252

## About The Pull Request
Administrators with a new permission flag can issue irrevocable, centcom-excluded bans.
This system is given a very high degree of protection befitting it's status as an incredibly dangerous tool.

## Why It's Good For The Game
Requested by Crossedfall.

## Changelog
:cl:
admin: +SUPPRESS can now be granted to allow access to issue hidden bans.
server: You should NOT be granting this lightly.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Why would you ever want a ban that is impossible to remove. I get supressed from centcom since the 18+ thing, but if someone appeals or there was a mistake its gg

**Comment by francinum** [comment]
> 
> 
> Why would you ever want a ban that is impossible to remove. I get supressed from centcom since the 18+ thing, but if someone appeals or there was a mistake its gg

If it's a mistake you ask Crossed.

**Comment by Rukofamicom** [comment]
> Why would you ever want a ban that is impossible to remove. 

Because age related bans are 100% unappealable. They are not, however, permanent and will still expire when appropriate.

Since they are totally unappealable anyway, and they involve sensitive player data, we don't need them publicly listed, and nobody except for the admin placing the ban and the player being banned needs to be aware of it.

**Comment by PowerfulBacon** [comment]
Age bans are absolutely appealable, time exists.

**Comment by ike709** [comment]
@PowerfulBacon The actual reason is GDPR Right to be Forgotten requests. We maintain the ban to operate our service but remove all references to the data outside of the ban DB itself. This feature already exists and is in use. This PR just expands the permissions.

Whether or not seniormins should have this ability is another story, but since permissions are configurable it is a non-issue w/ regards to merging this.

File: code/modules/admin/sql_ban_system.dm [hunk 1]
```diff
@@ -399,16 +417,22 @@
 	if(use_last_connection && !ip_check && !cid_check)
 		error_state += "Use last connection was ticked, but neither IP nor CID was."
 	if(href_list["applyadmins"])
+		if(redact)
+			error_state += "Admin bans can not not be suppressed."
```
Comment (ike709) [inline code/modules/admin/sql_ban_system.dm]: not not

**Comment by ike709** [comment]
Wtf is that DMI file change?

**Comment by PowerfulBacon** [comment]
> 
> 
> @PowerfulBacon The actual reason is GDPR Right to be Forgotten requests. We maintain the ban to operate our service but remove all references to the data outside of the ban DB itself. This feature already exists and is in use. This PR just expands the permissions.
> 
> Whether or not seniormins should have this ability is another story, but since permissions are configurable it is a non-issue w/ regards to merging this.

Yea but they still should be revokable. (by senior or headmins at least, rather than forcing database edits)

**Comment by Crossedfall** [comment]
> > @PowerfulBacon The actual reason is GDPR Right to be Forgotten requests. We maintain the ban to operate our service but remove all references to the data outside of the ban DB itself. This feature already exists and is in use. This PR just expands the permissions.
> > Whether or not seniormins should have this ability is another story, but since permissions are configurable it is a non-issue w/ regards to merging this.
> 
> Yea but they still should be revokable. (by senior or headmins at least, rather than forcing database edits)

If the bans were still made visible in a way that allowed them to be revoked, then it wouldn't abide by the strict standards set out by privacy guidelines and laws.

> Age bans are absolutely appealable, time exists.

That's not an appeal, that's just waiting out a temporary ban.

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Drop the badge

**Review by Crossedfall — APPROVED** [review]
I was looking for another reason to read this

**Comment by MCterra10** [comment]
Australia gang, no GDPR for us :trollface:

**Pull request merged**